src/t6/xsk/resolver.cpp

```diff
@@ -100,60 +100,27 @@ auto resolver::file_data(const std::string& name) -> std::tuple<const std::strin
     throw error("couldn't open gsc file '" + name + "'");
 }
 
-std::set<std::string_view> paths
-{
-    "aitype"sv,
-    "animscripts"sv,
-    "animscripts/ai_subclass"sv,
-    "animscripts/aitype"sv,
-    "animscripts/bigdog"sv,
-    "animscripts/traverse"sv,
-    "animscripts/turret"sv,
-    "character"sv,
-    "clientscripts"sv,
-    "clientscripts/mp"sv,
-    "clientscripts/mp/createfx"sv,
-    "clientscripts/mp/gametypes"sv,
-    "clientscripts/mp/zombies"sv,
-    "codescripts"sv,
-    "common_scripts"sv,
-    "maps"sv,
-    "maps/ai_subclass"sv,
-    "maps/createart"sv,
-    "maps/createfx"sv,
-    "maps/gametypes"sv,
-    "maps/mp"sv,
-    "maps/mp/animscripts"sv,
-    "maps/mp/animscripts/traverse"sv,
-    "maps/mp/bots"sv,
-    "maps/mp/createart"sv,
-    "maps/mp/createfx"sv,
-    "maps/mp/gametypes"sv,
-    "maps/mp/gametypes_zm"sv,
-    "maps/mp/killstreaks"sv,
-    "maps/mp/teams"sv,
-    "maps/mp/zombies"sv,
-    "maps/voice"sv,
-    "mpbody"sv,
-    "xmodelalias"sv,
-};
 
 auto resolver::fs_to_game_path(const std::filesystem::path& file) -> std::filesystem::path
 {
     auto result = std::filesystem::path("", std::filesystem::path::format::generic_format);
-    auto root = false;
+    std::string str;
+    if (file.is_relative())
+    {
+        str = file.generic_string();
+    }
+    else if(file.is_absolute())
+    {
+        str = std::filesystem::relative(file).generic_string();
+    }
 
-    for (auto& entry : file)
+    if (!str.empty())
     {
-        if (!root && paths.contains(entry.string()))
-        {
-            result = entry;
-            root = true;
-        }
-        else if (paths.contains(result.string()))
+        while(str.std::string::at(0) == '.' || str.std::string::at(0) == '/')
```

Comment:
i'm ass with C++ so go easy on me but is this right? i'm thinking it may be `str.at(index)` and not `str.std::string::at(index)`